libcudacxx/include/cuda/std/__bit/byteswap.h

```diff
@@ -37,37 +37,47 @@ _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
 class __byteswap_impl
 {
-  template <class _Half, class _Full>
+  template <class _Full>
   _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr _Full __impl_recursive(_Full __val) noexcept
   {
-    static_assert(sizeof(_Full) == sizeof(_Half) * 2, "Invalid half type passed to __bytswap_impl");
+    if constexpr (sizeof(_Full) / 2 > 1)
```

Comment:
That condition could be simpler 
```suggestion
    if constexpr (sizeof(_Full) > 2)
```

However, shouldnt this only be taken when the type is larger than 64 bits?